src/components/postcard/PostCard.js

```diff
@@ -0,0 +1,121 @@
+import { React, useState } from "react";
+import { useDispatch } from "react-redux";
+import { isPostLiked } from "../../features";
+import { useAuth } from "../../features/auth/authSlice";
+import {
+  addLikeToPost,
+  removeLikedPost,
+  usePosts,
+  setEditPostMode,
+  deletePost,
+} from "../../features/posts/postSlice";
+import { AddPostModal } from "../../modals";
+import { useNavigate } from "react-router-dom";
+
+export function PostCard({ post }) {
+  const navigate = useNavigate();
+  const [showModal, setShowModal] = useState(false);
+  const { isPostInEditMode } = usePosts();
+  const { user } = useAuth();
+  const dispatch = useDispatch();
+
+  const isPostAlreadyLiked = isPostLiked(user, post.likes);
+
+  const likePost = (id) => {
+    if (!isPostAlreadyLiked) {
```

Comment:
you can use ternary operator to make it more readable

Reply:
Done